src/components/file/file.repository.ts

```diff
@@ -93,20 +93,18 @@ export class FileRepository {
           createdBy: [{ id: 'createdById' }],
         },
       ])
-      .orderBy([`${input.sort} ${input.order}`])
-      .skip((input.page - 1) * input.count)
-      .limit(input.count);
-
+      .orderBy([`${input?.sort ?? 'name'} ${input?.order ?? 'ASC'}`])
+      .skip(((input?.page ?? 1) - 1) * (input?.count ?? 25))
+      .limit(input?.count ?? 25);
     const result = await query.run();
-
     const total = result.length === 0 ? 0 : result[0].total;
+    let children = result.map(baseNodeFromResult);
 
-    const children = result.map(baseNodeFromResult);
-
+    if (input?.filter?.type)
+      children = children.filter((item) => item.type === input.filter.type);
```

Comment:
This filter also needs to be applied in the query.
You can swap `FileNode` label on line 81 with the filter type if given.